src/main/java/sim/model/Server.java

```diff
@@ -6,36 +6,52 @@
 import sim.view.Observer;
 
 public class Server implements Runnable, Subject {
-    private SharedQueue customerQueue, kitchenQueue;
+    private SharedQueue customerQueue, kitchenQueue, priorityQueue;
     private Order[] currentOrder;
     private LinkedList<Observer> observers;
     private Logger log;
 
-    public Server(SharedQueue customerQueue, SharedQueue kitchenQueue) {
+    public Server(SharedQueue customerQueue, SharedQueue kitchenQueue, SharedQueue priorityQueue) {
         this.customerQueue = customerQueue;
         this.kitchenQueue = kitchenQueue;
+        this.priorityQueue = priorityQueue;
 
         observers = new LinkedList<Observer>();
         log = Logger.getInstance();
     }
 
+    public void threadAction(Order [] o) {
+        setCurrentOrder(o);
+        notifyObservers();
+        try {
+            // Time to process order depends on number of items
+            Thread.sleep(10000l * o.length);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+        }
+        log.add(o, Logger.OrderState.PROCESSED);
+        kitchenQueue.addOrder(o);
+    }
+
     @Override
     public void run() {
 
+        // Process queue from priorityQueue over customerQueue if there are still orders in it. 
         // Service continues as long as customers are still due to arrive or customers
-        // are in the queue
-        while (!customerQueue.getDone() || !customerQueue.isEmpty()) {
-            Order [] order = customerQueue.getCustomerOrder();
-            setCurrentOrder(order);
-            notifyObservers();
-        	try {
-                // Time to process order depends on number of items
-                Thread.sleep(10000l * order.length);
-            } catch (InterruptedException e) {
-                Thread.currentThread().interrupt();
+        // // are in the queue
+        while (!customerQueue.getDone() || !customerQueue.isEmpty()
+                || !priorityQueue.getDone() || !priorityQueue.isEmpty()) {
+            
+            // If orders still exist in the priority queue
+            if (!priorityQueue.getDone() || !priorityQueue.isEmpty()) {
```

Comment:
The logic here seems overly complicated (and incorrect), you're already in a while loop that's running while both lists aren't finished

This secondary while loop inside the main one will mean servers fully process the priority queue before even starting the regular queue

Reply:
Thinking about the common code between both queues, the only things that needed to change were the condition of the loop (so it runs until both are done) and the addition of an `if` statement to take from the priority queue first